src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java

```diff
@@ -144,4 +155,60 @@ public FileDownloadController(ExperimentFileLocationService experimentFileLocati
             zipOutputStream.close();
         }
     }
+
+    @GetMapping(value = "json/experiments/download/zip/check",
+            produces = "application/json;charset=UTF-8")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public String
+    checkMultipleExperimentsFileValid(@RequestParam(value = "accession", defaultValue = "") List<String> accessions)
+    {
+        List<Experiment> experiments = new ArrayList<>();
+
+        accessions.stream().forEach(accession -> {
+            try{
+                experiments.add(experimentTrader.getPublicExperiment(accession));
+            } catch (Exception e) {
+                LOGGER.debug("Invalid experiment accession: {}", accession);
+            }
+        });
+
+        ImmutableList<ExperimentFileType> fileTypeCheckList = ImmutableList.of(
+                ExperimentFileType.QUANTIFICATION_RAW,
+                ExperimentFileType.NORMALISED,
+                ExperimentFileType.EXPERIMENT_DESIGN);
+
+        var filePaths = experiments.stream()
```

Comment:
`filePaths` is of type `Map<String, ImmutableList<List<Path>>>`. Flattening the values so that it is `ImmutableMap<String, ImmutableList<Path>>` will make the filtering second block simpler. You only need a slight change in the mapping step below at L185.